src/frequenz/sdk/config/_manager.py

```diff
@@ -80,20 +95,53 @@ def __str__(self) -> str:
         return f"{type(self).__name__}[{self.name}]"
 
     # The noqa DOC502 is needed because we raise TimeoutError indirectly.
-    async def new_receiver(self) -> Receiver[Mapping[str, Any] | None]:  # noqa: DOC502
+    async def new_receiver(  # pylint: disable=too-many-arguments # noqa: DOC502
+        self,
+        *,
+        wait_for_first: bool = False,
```

Comment:
After the presentation of this to a wider audience I'm really considering to remove the *wait for first* **completely**, I think it adds a lot of complexity only to deal with a very niche an obscure situation where there might be a bug and for some reason we don't receive the first config.

I think we can improve debugability of that situation by adding a timeout to the reading of the config files in the config managing actor instead (if we don't have it yet), which would be probably the only place where this could really hang. Also we might add more logging before and after reading the first config, so if you see a "Waiting for first config" without an immediate "Got the initial config: %s", config, that would mean the config reading somehow got stuck. We can now do this in the ConfigManager itself, so users don't need to remember to add these log lines.

What do you think?

Reply:
I'm even more convinced we should remove this, as if we keep this option, the `new_receiver()` method needs to be `async`, and if it is `async`, it can be used in constructors, which complicates initialization even further.

Reply:
I finally moved it to an utility function. This removes all the complexity from the config manager, and if you need this functionality, you can just do a `await wait_for_first()` and that will get the default timeout if you don't pass one explicitly.